beginner_source/export_simple_model_to_onnx_tutorial.py

```diff
@@ -0,0 +1,150 @@
+    * ``torch.onnx.export`` is based on TorchScript backend and has been the default until PyTorch 2.0.
+
+In this tutorial, we describe how to convert a model defined in PyTorch into the ONNX format using
+the latest and preferred ``torch.onnx.dynamo_export` ONNX exporter.
+
+"""
+
+###############################################################################
+# In the `60 Minute Blitz <https://pytorch.org/tutorials/beginner/deep_learning_60min_blitz.html>`_,
+# we had the opportunity to learn about PyTorch at a high level and train a small neural network to classify images.
+#
+# While PyTorch is great for iterating on the development of models, the resulting models are not typically deployed
+# to production in this fashion. This is where `ONNX <https://onnx.ai/>`_ (Open Neural Network Exchange) comes in!
+# ONNX is a flexible open standard format for representing machine learning models which standardized representations
+# of machine learning that allow them to be executed across a gamut of hardware platforms and runtime environments
+# from large-scale cloud-based supercomputers to resource-constrained edge devices such as your web browser and phone.
+#
+# In this tutorial, we’ll learn how to:
+#
+# 1. Author a simple image classifier model in PyTorch (from the 60 Minute Blitz tutorial).
+# 2. Export the model to ONNX format.
+# 3. Save the ONNX model in a file.
+# 4. Visualize the ONNX model graph using `Netron <https://github.com/lutzroeder/netron>`_.
+# 5. Execute the ONNX model with `ONNX Runtime
+#
+# Note that because the ONNX exporter uses ``onnx`` and ``onnxscript`` to translate PyTorch operators into ONNX operators,
+# we will need to install them.
+# %%
+#  .. code-block:: bash
+#
+#      %%bash
+#      pip install onnx
+#      pip install onnxscript
+#
+# Once your environment is set up, let’s start modeling our image classifier with PyTorch,
+# exactly like we did in the 60 Minute Blitz tutorial.
+#
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class Net(nn.Module):
+
+    def __init__(self):
+        super(Net, self).__init__()
+        self.conv1 = nn.Conv2d(1, 6, 5)
+        self.conv2 = nn.Conv2d(6, 16, 5)
+        self.fc1 = nn.Linear(16 * 5 * 5, 120)
+        self.fc2 = nn.Linear(120, 84)
+        self.fc3 = nn.Linear(84, 10)
+
+    def forward(self, x):
+        x = F.max_pool2d(F.relu(self.conv1(x)), (2, 2))
+        x = F.max_pool2d(F.relu(self.conv2(x)), 2)
+        x = torch.flatten(x, 1)
+        x = F.relu(self.fc1(x))
+        x = F.relu(self.fc2(x))
+        x = self.fc3(x)
+        return x
+
+net = Net()
+
+# Analogous to the 60 Minute Blitz tutorial, we need to create a random 32x32 input.
+
+input = torch.randn(1, 1, 32, 32)
+
+# That is all we need to export the model to ONNX format: a model instance and a dummy input.
+# We can now export the model with the following code:
+
+export_output = torch.onnx.dynamo_export(net, input)
```

Comment:
does this work if batch size changes? or more generally a quick comment on input shape changes would be helfpul

Reply:
I was planning to extend this tutorial with "advanced features" such as the `ExportOptions.dynamic_shapes` and friends.